File: cpp/leetcode/092.cpp
class Solution {  
public:
    ListNode* reverseBetween(ListNode* head, int m, int n) {
        ListNode* nhead = new ListNode(0);
        nhead->next = head;
        ListNode* ptr = nhead;
        for (int i = 0; i < m - 1; i++) ptr = ptr -> next;
        ListNode* start = ptr -> next;
        for (int i = 0; i < n - m; i++) {
            ListNode* end = start -> next; 
            start -> next = end -> next;
            end -> next = ptr -> next;
            ptr -> next = end;
        }
        return nhead -> next;
    }
};

File: python/leetcode/092.py
class Solution:
    def reverseBetween(self, head, m, n):
        if not head or not head.next: return head
        dummy = ListNode(0)
        dummy.next, ptr = head, dummy
        for i in range(m - 1): ptr = ptr.next
        start, end = ptr.next, ptr.next.next
        for i in range(n - m):  # reverse a pair node each time
            start.next = end.next
            end.next = ptr.next
            ptr.next = end
            end = start.next
        return dummy.next